OpenRA.Mods.Common/Graphics/DefaultSpriteSequence.cs

```diff
@@ -261,27 +261,33 @@ public DefaultSpriteSequence(ModData modData, TileSet tileSet, SpriteCache cache
 						var subOffset = LoadField(sd, "Offset", float3.Zero);
 						var subFlipX = LoadField(sd, "FlipX", false);
 						var subFlipY = LoadField(sd, "FlipY", false);
+						var subFrames = LoadField<int[]>(sd, "Frames", null);
 						var subLength = 0;
 
 						Func<int, IEnumerable<int>> subGetUsedFrames = subFrameCount =>
 						{
 							MiniYaml subLengthYaml;
 							if (sd.TryGetValue("Length", out subLengthYaml) && subLengthYaml.Value == "*")
-								subLength = subFrameCount - subStart;
+								subLength = subFrames != null ? subFrames.Length : subFrameCount - subStart;
 							else
 								subLength = LoadField(sd, "Length", 1);
 
-							return Enumerable.Range(subStart, subLength);
+							var subUsedFrames = new List<int>();
```

Comment:
This can simplify to `return subFrames ?? Enumerable.Range(subStart, subLength);`